$Header: /var/cvsroot/gentoo/src/patchsets/seamonkey/1.0.3/Attic/070_mozilla-mms-rm-fix.patch,v 1.1 2006/07/30 11:03:30 redhatter Exp $

This patch fixes an issue with mms:// and rm:// URLs used by the Windows Media
Player and RealOne Player respectively.  Without this patch, plugins such as the
MPlayer plugin would have difficulty displaying some embedded videos.

See https://bugzilla.mozilla.org/show_bug.cgi?id=346167 for more information.
Index: content/base/src/nsNoDataProtocolContentPolicy.cpp
===================================================================
RCS file: /cvsroot/mozilla/content/base/src/nsNoDataProtocolContentPolicy.cpp,v
retrieving revision 3.1
diff -u -8 -p -r3.1 nsNoDataProtocolContentPolicy.cpp
--- content/base/src/nsNoDataProtocolContentPolicy.cpp	26 Jun 2006 17:57:42 -0000	3.1
+++ content/base/src/nsNoDataProtocolContentPolicy.cpp	27 Jul 2006 22:14:40 -0000
@@ -63,18 +63,17 @@ nsNoDataProtocolContentPolicy::ShouldLoa
                                           nsISupports *aExtra,
                                           PRInt16 *aDecision)
 {
   *aDecision = nsIContentPolicy::ACCEPT;
 
   if (aContentType == TYPE_OTHER ||
       aContentType == TYPE_SCRIPT ||
       aContentType == TYPE_IMAGE ||
-      aContentType == TYPE_STYLESHEET ||
-      aContentType == TYPE_OBJECT) {
+      aContentType == TYPE_STYLESHEET) {
     nsCAutoString scheme;
     aContentLocation->GetScheme(scheme);
     // Fast-track for the common cases
     if (scheme.EqualsLiteral("http") ||
         scheme.EqualsLiteral("https") ||
         scheme.EqualsLiteral("ftp") ||
         scheme.EqualsLiteral("file") ||
         scheme.EqualsLiteral("chrome")) {
